lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -1063,32 +872,39 @@ def alerts(self) -> dict:
                 except KeyError:
                     logger.error("Alert rules were found but no usable labels were present")
 
+            _, errmsg = self.tool.validate_alert_rules(alert_rules)
+            if errmsg:
+                if alerts[identifier]:
+                    del alerts[identifier]
+                relation.data[self._charm.app]["event"] = json.dumps({"errors": errmsg})
+                continue
+
         return alerts
 
 
 # Copy/pasted from prometheus_scrape.py
-class PromqlTransformer:
-    """Uses promql-transform to inject label matchers into alert rule expressions."""
+class CosTool:
+    """Uses cos-tool to inject label matchers into alert rule expressions and validate rules."""
```

Comment:
No. It's utility wrappers around the same binary.

`_charm` doesn't seem to be used, but since `PromqlTransform` used it, I left it until we find out whether there was a reason for htis.